### ISSUE TITLE
IDocumentProcessor Issue introduced between 11.18.2 and 11.18.3 

### ISSUE THREAD

**Issue by wi937c-amr** [RSuter/NSwag#1530]
On NSwag.AspNetCore 11.18.2 the following document processor will correctly add the http and https schemes to the the swagger document

```
 public class EnableHttpsSchemeDocumentProcessor : IDocumentProcessor
    {
        Task IDocumentProcessor.ProcessAsync(DocumentProcessorContext context)
        {
            var schemes = context.Document.Schemes;
            if (!schemes.Contains(NSwag.SwaggerSchema.Http)) { schemes.Add(NSwag.SwaggerSchema.Http); }
            if (!schemes.Contains(NSwag.SwaggerSchema.Https)) { schemes.Add(NSwag.SwaggerSchema.Https); }
            return Task.CompletedTask;
        }
    }
```

The resulting document is as follows in 11.18.2

```
{
  "x-generator": "NSwag v11.18.2.0 (NJsonSchema v9.10.67.0 (Newtonsoft.Json v11.0.0.0))",
  "swagger": "2.0",
  "info": {
    "title": "MyAPI",
    "version": "1.0.0"
  },
  "host": "dev-api.example.com",
  "schemes": [
    "http",
    "https"
  ],
...
```

After updating to 11.18.3 or after

```
{
  "x-generator": "NSwag v11.18.3.0 (NJsonSchema v9.10.67.0 (Newtonsoft.Json v11.0.0.0))",
  "swagger": "2.0",
  "info": {
    "title": "MyAPI",
    "version": "1.0.0"
  },
  "host": "dev-api.example.com",
  "schemes": [
    "http"
  ],
...
```

rolling back to 11.18.2  produces correct result

**Comment by RSuter** [comment]
Regression of this commit: https://github.com/RSuter/NSwag/commit/663b8bd11d41eb904f7db132765508bb6c019dec

**Comment by RSuter** [comment]
Fixed in v11.18.5

**Comment by wi937c-amr** [comment]
Thank you for the quick response.  It now works as expected.
